components/js_lang/js_lang_core.cc

```diff
@@ -569,15 +573,42 @@ bool Js_sp::execute() {
     }
   }
 
-  // Call the wrapper function passing parameters to it.
+  /*
+    We about to execute JS code, which can take time. Make this execution
+    KILLable by setting killable_isolate flag, which indicates that one
+    needs to call Isolate::TerminateExecution() to abort execution of JS
+    code in the connection as well indicates Isolate for which this method
+    needs to be called.
+  */
+  js_thd->set_killable_isolate(isolate);
+
+  // But before proceeding, we need to check if we have been killed
+  // before we set killable_isolate and have not noticed this.
+  bool killed_earlier = js_thd->is_killed();
+
+  v8::MaybeLocal<v8::Value> maybe_result;
+  if (!killed_earlier) {
+    // Call the wrapper function passing parameters to it.
+    maybe_result =
+        func->Call(context, context->Global(), m_arg_count, arg_arr);
+  }
+
+  js_thd->set_killable_isolate(nullptr);
+  delete[] arg_arr;
```

Comment:
May be some kind of RAII object for `arg_arr`? You already do `delete[] arg_arr` earlier.

Reply:
Done.